recipes/yojimbo/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+from conans import ConanFile, CMake, MSBuild, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class YojimboConan(ConanFile):
+    name = "yojimbo"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/networkprotocol/yojimbo"
+    topics = ("conan", "yojimbo", "game", "udp", "protocol", "client-server", "multiplayer-game-server")
+    description = "A network library for client/server games written in C++"
+    license = "BSD-3-Clause"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    build_requires = "premake/5.0.0-alpha15"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"fPIC": [True, False]}
+    default_options = {"fPIC": True}
+
+    _source_subfolder = "source_subfolder"
+    _build_subfolder = "build_subfolder"
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def requirements(self):
+        self.requires("libsodium/1.0.18")
+        self.requires("mbedtls/2.25.0")
+ 
+    def source(self):
+        git = tools.Git(folder=self._source_subfolder)
+        git.clone(self.homepage, branch="v"+self.version, shallow=True)
+        git.run("submodule update --init --recursive --recommend-shallow")
```

Comment:
We typically do not work this way on CCI, you can check the docs https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md#the-conandatayml

It's in part why the build is failing.

You'll need to add the two subprojects as their own recipes first, that's been our rule thus far